android/src/main/java/com/robinpowered/react/ScreenBrightness/ScreenBrightnessModule.java

```diff
@@ -3,52 +3,241 @@
+     * This method is callable from the JS context.
+     */
+    @ReactMethod
+    public void requestPermission() {
+        requestSettingsPermission();
+    }
+
+    /**
+     * Updates the device brightness.
+     * This method is callable from the JS context.
+     *
+     * @param brightness The brightness level between 0-1.
+     * @param promise A promise resolving if the brightness was updated.
+     */
+    @ReactMethod
+    public void setSystemBrightness(float brightness, final Promise promise) {
+        if (setSystemBrightness((int) (brightness * BRIGHTNESS_MAX))) {
+            promise.resolve(brightness);
+        } else {
+            promise.reject(null);
+        }
+    }
+
+    /**
+     * Gets the brightness level of the device.
+     * This method is callable from the JS context.
+     *
+     * @param promise A promise resolving the brightness level.
+     */
+    @ReactMethod
+    public void getSystemBrightness(final Promise promise) {
+        promise.resolve(getSystemBrightness());
+    }
+
+    /**
+     * Gets the application brightness level
+     * This method is callable from the JS context.
+     *
+     * @param promise A promise resolving the app brightness level.
+     */
+    @ReactMethod
+    public void getAppBrightness(Promise promise) {
+        float brightness = activity.getWindow().getAttributes().screenBrightness;
         promise.resolve(brightness);
     }
 
+    /**
+     * Sets the application brightness level.
+     * This method is callable from the JS context.
+     *
+     * @param brightness The brightness level.
+     * @param promise A promise resolving the updated brightness level.
+     */
     @ReactMethod
-    public void setBrightness(final float brightness, final Promise promise) {
-        mActivity.runOnUiThread(new Runnable() {
+    public void setAppBrightness(final float brightness, final Promise promise) {
+        activity.runOnUiThread(new Runnable() {
             @Override
             public void run() {
-                WindowManager.LayoutParams lp = mActivity.getWindow().getAttributes();
-                lp.screenBrightness=brightness;
-                mActivity.getWindow().setAttributes(lp);
+                WindowManager.LayoutParams lp = activity.getWindow().getAttributes();
+                lp.screenBrightness = brightness;
+                activity.getWindow().setAttributes(lp);
                 promise.resolve(brightness);
             }
         });
     }
-}
\ No newline at end of file
+
+    /**
+     * Gets the brightness level of the device.
+     * This method is callable from the JS context.
+     *
+     * @deprecated
+     * @param promise A promise resolving the brightness level.
+     */
+    @ReactMethod
+    public void getBrightness(final Promise promise) {
+        getSystemBrightness(promise);
+    }
+
+    /**
+     * Updates the device brightness.
+     * This method is callable from the JS context.
+     *
+     * @deprecated
```

Comment:
These should read something like:

`@deprecated Use {@link #getSystemBrightness()} instead.`